src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -247,6 +247,7 @@ private List<MetadataReference> GetApplicationReferences()
                 return metadataReferences;
             }
 
+            var libraryPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Can you file an issue to make sure we don't hold onto the dependency context forever?